From cdac9f30769e2028d258d159485d380202a3cec7 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 14 Apr 2016 14:42:29 +0300
Subject: [PATCH 115/239] fix: tools: a3700: Take clock preset from defconfig

- Use u-boot defconfig instaed of DT for extracting
  the CPU/DDR clock preset.
- Fixes SYSTEMSW-2433

Change-Id: I3af9633efd74e121a39ed805fd91a4d2f9e68f1e
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29050
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 36b8ed0..ae7ac3f 100644
--- a/Makefile
+++ b/Makefile
@@ -363,7 +363,6 @@ ifeq (${PLAT},a3700_z)
 DOIMAGEPATH		?= tools/wtp
 DOIMAGETOOL		?= $(shell which ntbb_linux.exe)
 
-DDRCFG			:= $(DOIMAGEPATH)/ddr-600.txt
 TIMBUILD		:= $(DOIMAGEPATH)/buildtim.sh
 TIM2IMG			:= $(DOIMAGEPATH)/tim2img.pl
 DOIMAGE_CFG		:= $(DOIMAGEPATH)/atf-ntim.txt
@@ -378,8 +377,8 @@ CONFIG_MVEBU_SPI_BOOT	:= $(shell grep "CONFIG_MVEBU_SPI_BOOT" -m 1 $(SRCPATH)/.c
 CONFIG_MVEBU_MMC_BOOT	:= $(shell grep "CONFIG_MVEBU_MMC_BOOT" -m 1 $(SRCPATH)/.config |cut -d"=" -f2)
 
 CLOCKSPATH		:= $(DOIMAGEPATH)
-CLOCKSPRESET		:= $(shell find $(SRCPATH) -name $(CONFIG_DEFAULT_DEVICE_TREE).dts | \
-				xargs grep preset | sed -n 's/.*<\([^ ]*\)>;/\1/p')
+CLOCKSPRESET		:= $(shell grep CONFIG_PRESET_ $(SRCPATH)/.config |grep =|cut -d"=" -f1|cut -c 8-)
+
 ifeq ($(CONFIG_MVEBU_SPI_BOOT),y)
 BOOTDEV			:= SPINOR
 PARTNUM			:= 0
-- 
1.9.1

